### ISSUE TITLE
BRJS could be more friendly when leftover test folders are present

### ISSUE THREAD

**Issue by janhancic** [BladeRunnerJS/brjs#1302]
Let's say I have a library (or a blade, bladeset, ...) that uses the "old" folder structure for tests:

```
tests/
  test-unit/
    js-test-driver/
      ...
```

I then decide to use the new simplified structure and move my tests to the new structure:

```
test-unit/
  ...
```

I do this with git and commit everything and everything is fine.

But then, another developer pulls my changes (it also often happens on build machines). Because Git doesn't clean down the folders properly (because there are the `bundles` folder in the old location which aren't tracked by Git), BRJS then fails to run the tests for that library, which is fair enough, it expects one location for tests.

But it errors out with this exception:

```
Testing fooapp\libs\foo-lib\test-unit (UTs):
Unexpected Exception:
java.lang.ClassCastException: org.bladerunnerjs.model.AppJsLib cannot be cast to org.bladerunnerjs.model.TestPack
        at com.caplin.cutlass.command.test.testrunner.JsTestDriverBundleCreator.createRequiredBundles(JsTestDriverBundleCreator.java:43)
        at com.caplin.cutlass.command.test.testrunner.TestRunner.runTest(TestRunner.java:368)
        at com.caplin.cutlass.command.test.testrunner.TestRunner.runTestAndRecordDuration(TestRunner.java:354)
        at com.caplin.cutlass.command.test.testrunner.TestRunner.runAllTestsInDirectory(TestRunner.java:325)
        at com.caplin.cutlass.command.test.testrunner.TestRunner.runTests(TestRunner.java:160)
        at com.caplin.cutlass.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:118)
        at com.caplin.cutlass.command.test.TestCommand.doCommand(TestCommand.java:55)
        at org.bladerunnerjs.utility.CommandRunner.run(CommandRunner.java:21)
        at org.bladerunnerjs.model.BRJS.runCommand(BRJS.java:354)
        at org.bladerunnerjs.utility.UserCommandRunner.doRunCommand(UserCommandRunner.java:35)
        at org.bladerunnerjs.utility.UserCommandRunner.run(UserCommandRunner.java:27)
        at org.bladerunnerjs.model.BRJS.runUserCommand(BRJS.java:359)
        at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:104)
        at org.bladerunnerjs.runner.CommandRunner.main(CommandRunner.java:55)
```

Which isn't helpful to a developer trying to understand what's wrong.

Could BRJS provide a more useful error message, alterting the developers that he has the old folders?

Note that it's enough to have the following folder structure, for this to happen:

```
src/
test-unit/
  ... the actual tests ...
tests/
```

**Comment by ioanalianabalas** [comment]
Fixed by #1436.

**Comment by andyberry88** [comment]
I've created #1446 which supersedes #1436.

- A warning will be logged if both `tests` and `test-*` dirs exist.
- If both exist and `tests` is empty `test-*` will be used, otherwise we revert to using `tests`.

This should mean leftover `tests` dirs are ignore entirely but we can still support the `tests` directory if it's intentionally used.
